lib/plugins/package/lib/zipService.js

```diff
@@ -131,53 +145,39 @@ function excludeNodeDevDependencies(servicePath) {
     });
 
     // filter out non node_modules file paths
-    const relevantFilePaths = _.filter(packageJsonFilePaths, (filePath) => {
+    const packageJsonPaths = _.filter(packageJsonFilePaths, (filePath) => {
       const isNodeModulesDir = !!filePath.match(/node_modules/);
       return !isNodeModulesDir;
     });
 
-    _.forEach(relevantFilePaths, (relevantFilePath) => {
+    // NOTE: using mapSeries here for a sequential computation (w/o race conditions)
+    return BbPromise.mapSeries(packageJsonPaths, (packageJsonPath) => {
       // the path where the package.json file lives
-      const fullPath = path.join(servicePath, relevantFilePath);
-      const rootDirPath = fullPath.replace(path.join(path.sep, 'package.json'), '');
-
-      process.chdir(rootDirPath);
-
-      // TODO replace with package-manager independent directory traversal?!
-      const prodDependencies = childProcess
-        .execSync('npm ls --prod=true --parseable=true --silent')
-        .toString().trim();
-
+      const fullPath = path.join(servicePath, packageJsonPath);
+      const dirWithPackageJson = fullPath.replace(path.join(path.sep, 'package.json'), '');
+
+      return childProcess.execAsync(
+        `npm ls --dev=true --parseable=true --silent >> ${nodeDevDepFile}`,
+        { cwd: dirWithPackageJson }
+      );
+    })
+    .then(() => fs.readFileAsync(nodeDevDepFile))
+    .then((fileContent) => {
+      const dependencies = fileContent.toString('utf8').split('\n');
```

Comment:
Shouldn't this be deduplicated (as it might be a combination of multiple package.json files and could contain duplicates)?
```
const dependencies = _.uniq(_.split(fileContent.toString('utf8'), '\n'));
```

I also suggest to use lodash, as otherwise you have to test here that fileContent is not nil, before using toString().